### ISSUE TITLE
Release 1.23.0 requires min API 21

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-Android#1005]
The dependency upgrade of `okhttp` from `3.12.1` to `3.14.6` makes the Parse SDK require min API level 21.

I think this is important to note and should be mentioned at least as a release note.
The library should also require `minSdkVersion = 21`, it currently seems to require 14.

https://github.com/parse-community/Parse-SDK-Android/blob/e9d986f86297e499d307aeb5a73ab88cf20a7c4a/build.gradle#L33

- https://medium.com/square-corner-blog/okhttp-3-13-requires-android-5-818bb78d07ce
- https://github.com/square/okhttp/issues/4746

Crash log looks like this, when adding the Parse SDK to a project with a lower API than 21:

```
okhttp3.OkHttpClient.newSslSocketFactory (OkHttpClient.java:263)
okhttp3.OkHttpClient.<init> (OkHttpClient.java:229)
okhttp3.OkHttpClient$Builder.build (OkHttpClient.java:1015)
com.parse.ParseHttpClient.<init> (ParseHttpClient.java:46)
com.parse.ParseHttpClient.createClient (ParseHttpClient.java:50)
com.parse.ParsePlugins.restClient (ParsePlugins.java:145)
com.parse.ParseCorePlugins.getConfigController (ParseCorePlugins.java:224)
com.parse.ParseConfig.getConfigController (ParseConfig.java:42)
com.parse.ParseConfig.getCurrentConfig (ParseConfig.java:54)
com.parse.Parse$2.then (Parse.java:183)
com.parse.Parse$2.then (Parse.java:179)
bolts.Task$14.run (Task.java:872)
```

**Comment by Jawnnypoo** [comment]
Ah, this was an oversight. I was thinking that 4.0.0 was the version that started requiring 5.0+. It will be corrected soon.

**Comment by Jawnnypoo** [comment]
Fixed with 1.23.1, thanks again for reporting!

**Comment by mtrezza** [comment]
@Jawnnypoo Thanks for the quick fix 🚀
